tests/Unit/NumericalAlgorithms/LinearOperators/Test_CoefficientTransforms.cpp

```diff
@@ -0,0 +1,133 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/IndexIterator.hpp"
+#include "DataStructures/ModalVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"
+#include "NumericalAlgorithms/LinearOperators/CoefficientTransforms.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Algorithm.hpp"
+#include "Utilities/Gsl.hpp"
+
+// These tests generate a function `u_expected` from a linear superposition of
+// the basis functions, which are then transformed to spectral space
+// (u_coeff) and the coefficients compared with the coefficients set in physical
+// space. The functions are then transformed back (u_back) and compared to the
+// original function, `u_expected`.
```

Comment:
I don’t understand what you mean with „physical space“ here. „..transformed to spectral space and the modal coefficients compared to the modes from which the function was constructed“ perhaps?

Having this comment split over the lines in the `check_transforms` function (i.e. narrating it) would make things more clear I think (up to you).